builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -854,6 +857,10 @@ func resourceArmVirtualMachineStorageOsDiskHash(v interface{}) int {
 	if m["vhd_uri"] != nil {
 		buf.WriteString(fmt.Sprintf("%s-", m["vhd_uri"].(string)))
 	}
+	if encryptionSettings := m["encryption_settings"]; encryptionSettings != nil {
+		var encryptionHash = schema.HashSchema(encryptionSettingsSchema())(encryptionSettings)
```

Comment:
It doesn't have a hash function, on purpose (going by the guidance of not using one for new code, to avoid the issues of when people add members but forget to update the code).